DeleteCgnOrchestrator/handler.ts

```diff
@@ -0,0 +1,299 @@
+          E.getOrElseW(e =>
+            trackExAndThrowWithError(
+              e,
+              "cgn.delete.exception.decode.eycaExpirationOutput"
+            )
+          )
+        );
+        if (decodedDeleteEycaExpirationResult.kind !== "SUCCESS") {
+          trackExAndThrowWithError(
+            new Error("Cannot delete EYCard Expiration data"),
+            "cgn.delete.exception.failure.eycaExpirationActivityOutput"
+          );
+        }
+
+        // Delete Eyca Card
+        const deleteEycaResult = yield context.df.callActivityWithRetry(
+          "DeleteEycaActivity",
+          internalRetryOptions,
+          DeleteEycaActivityInput.encode({ fiscalCode })
+        );
+        const decodedDeleteEycaResult = pipe(
+          deleteEycaResult,
+          DeleteEycaActivityResult.decode,
+          E.getOrElseW(e =>
+            trackExAndThrowWithError(
+              e,
+              "cgn.delete.exception.decode.eycaOutput"
+            )
+          )
+        );
+        if (decodedDeleteEycaResult.kind !== "SUCCESS") {
+          trackExAndThrowWithError(
+            new Error("Cannot delete EYCard"),
+            "cgn.delete.exception.failure.eycaActivityOutput"
+          );
+        } else {
+          eycaDataToBackup = decodedDeleteEycaResult.cards;
+        }
+      }
+
+      // Delete Cgn Expiration Data
+      const deleteCgnExpirationResult = yield context.df.callActivityWithRetry(
+        "DeleteCgnExpirationActivity",
+        internalRetryOptions,
+        DeleteCgnExpirationActivityInput.encode({ fiscalCode })
+      );
+      const decodedDeleteCgnExpirationResult = pipe(
+        deleteCgnExpirationResult,
+        ActivityResult.decode,
+        E.getOrElseW(e =>
+          trackExAndThrowWithError(
+            e,
+            "cgn.delete.exception.decode.cgnExpirationOutput"
+          )
+        )
+      );
+      if (decodedDeleteCgnExpirationResult.kind !== "SUCCESS") {
+        trackExAndThrowWithError(
+          new Error("Cannot delete CGN expiration data"),
+          "cgn.delete.exception.failure.cgnExpirationActivityOutput"
+        );
+      }
+
+      // Delete Cgn Card
+      const deleteCgnResult = yield context.df.callActivityWithRetry(
+        "DeleteCgnActivity",
+        internalRetryOptions,
+        DeleteCgnActivityInput.encode({ fiscalCode })
+      );
+      const decodedDeleteCgnResult = pipe(
+        deleteCgnResult,
+        DeleteCgnActivityResult.decode,
+        E.getOrElseW(e =>
+          trackExAndThrowWithError(e, "cgn.delete.exception.decode.cgnOutput")
+        )
+      );
+      if (decodedDeleteCgnResult.kind !== "SUCCESS") {
+        trackExAndThrowWithError(
+          new Error("Cannot delete CGN"),
+          "cgn.delete.exception.failure.cgnActivityOutput"
+        );
+      } else {
+        cgnDataToBackup = decodedDeleteCgnResult.cards;
+
+        // Backup all data for legal issue
+        const legalDataToBackup: DeleteLegalDataBackupActivityInput = {
```

Comment:
What if eyca is deleted and an error is throw by cgn deletion? Will eyca data be recoverable?